### PR TITLE
fix(TTML): Fix duplicate cues overlapping segment boundaries

### LINKED ISSUE #4631: EBU subtitles regression in 4.1.3 - duplicate line on screen
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
yes

**What link can we use to reproduce this?**
https://shaka-player-demo.appspot.com/demo/#audiolang=en;textlang=en;uilang=en;asset=https://otvplayer.nagra.com/demo/content/bbc/bbc_ebu-tt-d_subtitle_fragmented/dash/ondemand/elephants_dream/1/client_manifest-all-minus-1080p.mpd;panel=CUSTOM%20CONTENT;build=uncompiled

**What version of Shaka Player are you using?**
v4.2.2-uncompiled

also seen in 4.2.3, 4.2.1, 4.2.0, 4.1.5, 4.1.4, 4.1.3

Its OK in 4.1.2

**What browser and OS are you using?**
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36

**What did you do?**

1. Play this stream
2. enable captions
3. monitor the text that is rendered

**What did you expect to happen?**
At around 3 mins we should see the line
`You - I mean we - we could easily die out here`
There are several other examples of duplication in the text during this stream, this is just one example.

**What actually happened?**
The line is repeated.  Two copies.

image

### PR DESCRIPTION
**Pull request by joeyparrish**
Closes #4631

### CODE REVIEW

**Review by theodab — APPROVED** [review]

**Review by littlespex — APPROVED** [review]

**Comment by joeyparrish** [comment]
Oops, had a linter error, and fixing it dismissed your reviews.  Please look again when you have a chance.

**Review by littlespex — APPROVED** [review]

**Comment by joeyparrish** [comment]
Waiting to confirm with @david-hm-morgan before merging, since this clobbers his PR.

**Review by avelad — APPROVED** [review]

**Review by david-hm-morgan — APPROVED** [review]
Yes this is a more suitable solution, I've retested with the original problematic stream and all is good.
thanks @joeyparrish

**Pull request merged**
